### PR TITLE
Added middleware for extension request for /request route

### PR DESCRIPTION
**Pull request by sahsisunny**
## Issue Ticket Number

- #2019

## Description

In the PR, a new middleware was added to create extension requests through the request router. This middleware will facilitate the creation of extension requests for the user.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [x] Yes
- [ ] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

image

image

</details>

## Test Coverage

| File                        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s |
|-----------------------------|---------|----------|---------|---------|-------------------|
| extensionRequestsv2.ts      |   100   |   100    |   100   |   100   |                   |
| requests.ts                 | 87.75   | 44.44    |   100   | 85.71   | 18,27-30,50,59    |

## Additional Notes

In the next PR, a controller will be added to handle the extension request creation.

### CODE REVIEW

File: test/unit/middlewares/extensionRequests.test.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
+import chai from "chai";
+import sinon from "sinon";
+const { expect } = chai;
+
+import { createExtensionRequestValidator } from "../../../middlewares/validators/extensionRequestsv2";
+import { extensionCreateObject } from "../../fixtures/extension-requests/extensionRequests";
+import { REQUEST_STATE } from "../../../constants/requests";
+
+describe("Extension Request Validators", function () {
```
Comment (tejaskh3) [inline test/unit/middlewares/extensionRequests.test.ts:9]: I Have this as a question in my mind, as I don't know much about testing. why is it working fine??

What I mean is you have used the request type "ExtensionRequestRequest" in side the function, but while testing we are using req type any. shouldn't it trow an error for expecting req type "ExtensionRequestRequest" but got any.

**Review by tejaskh3 — APPROVED** [review]
LGTM

File: test/unit/middlewares/extensionRequests.test.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    let req: any;
```
Comment (tejaskh3) [inline test/unit/middlewares/extensionRequests.test.ts:10]: I Have this as a question in my mind, as I don't know much about testing. why is it working fine??

What I mean is you have used the request type "ExtensionRequestRequest" in side the function, but while testing we are using req type any. shouldn't it trow an error for expecting req type "ExtensionRequestRequest" but got any.

File: test/unit/middlewares/extensionRequests.test.ts [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        nextSpy = sinon.spy();
+    });
+
+    describe("createExtensionRequestValidator", function () {
+        it("should validate for a valid create extension request", async function () {
+            req = {
+                body: extensionCreateObject,
+            };
+            res = {};
+            console.log("req", req);
```
Comment (skv93-coder) [inline test/unit/middlewares/extensionRequests.test.ts:29]: Can you please console.log

File: test/unit/middlewares/extensionRequests.test.ts (hunk 2 above)
Comment (sahsisunny) [reply test/unit/middlewares/extensionRequests.test.ts]: Done, added type for req and res

File: test/unit/middlewares/extensionRequests.test.ts (hunk 3 above)
Comment (sahsisunny) [reply test/unit/middlewares/extensionRequests.test.ts]: removed

**Review by skv93-coder — APPROVED** [review]

**Pull request merged**